src/main/scala/org/bitcoins/core/util/Base58.scala

```diff
@@ -0,0 +1,80 @@
+package org.bitcoins.core.util
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success}
+
+/**
+  * Created by chris on 5/16/16.
+  * source of values: https://en.bitcoin.it/wiki/Base58Check_encoding
+  */
+trait Base58 extends BitcoinSLogger {
+
+  val base58Characters = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+  val base58Pairs = base58Characters.zipWithIndex.toMap
+
+
+  def decodeCheck(input: String) : Seq[Byte] = {
+    val decoded : Seq[Byte] = decodeBase58(input)
+    if (decoded.length < 4) throw new RuntimeException("Invalid input")
+    else {
+      val splitSeqs = decoded.splitAt(decoded.length - 4)
+      val data : Seq[Byte] = splitSeqs._1
+      val checksum : Seq[Byte] = splitSeqs._2
+      val actualChecksum : Seq[Byte] = CryptoUtil.doubleSHA256(data).slice(0, 4)
+      if (checksum == actualChecksum)
+      data
+      else throw new IllegalArgumentException("checksums don't validate")
+    }
+  }
+
+
+  /**
+    * Takes in sequence of bytes and returns base58 bitcoin address
+    * Used ACINQ's implementation as reference under Apache License
+    * Modified to use Scala's BigInt rather than Java's BigInteger
+    * https://github.com/ACINQ/bitcoin-lib/blob/master/src/main/scala/fr/acinq/bitcoin/Base58.scala
+    *
+    * @param bytes
+    * @return
+    */
+  def encodeBase58(bytes : Seq[Byte]) : String = {
+    if (bytes.isEmpty) ""
+    else {
+      val big : BigInt = BigInt(1, bytes.toArray)
+      val builder = new StringBuilder
+
+      @tailrec
+      def loop(current : BigInt) : String = current match {
+        case a if current == BigInt(0) => ""
+        case _ =>
```

Comment:
Instead of case _ => 

it is good practice to put a type on the wildcard aka

case _ : BigInt => ...